isacc_messaging/api/isacc_record_creator.py

```diff
@@ -422,13 +434,49 @@ def execute_requests(self) -> Tuple[List[dict], List[dict]]:
 
         for cr_json in next_in_bundle(result):
             cr = CommunicationRequest(cr_json)
+            occurrence_utc = cr.occurrenceDateTime.date.replace(tzinfo=timezone.utc)
+            patient = resolve_reference(cr.recipient[0].reference)
+
+            # Happens when the patient removes their phone number completely.
+            # Should not occur in production.
+            try:
+                patient_unsubscribed = any(
+                    telecom_entry.system.lower() == 'sms' and telecom_entry.period.end 
+                    for telecom_entry in patient.telecom
+                )
+            except Exception as e:
+                skipped_crs.append(cr)
+                audit_entry(
+                    f"Failed to send the message, {patient} does not have phone number",
+                    extra={"resource": f"CommunicationResource/{cr.id}", "exception": e},
+                    level='exception'
+                )
+                # Display Twilio Error in a human readable form
+                errors.append({'id': cr.id, 'error': str(e)})
+                continue
+
             if cr.occurrenceDateTime.date < cutoff:
-                # anything older than cutoff will never be sent (#1861758)
+                # Anything older than cutoff will never be sent (#1861758)
                 # and needs a status adjustment lest it throws off other queries
                 # like next outgoing message time
                 skipped_crs.append(cr)
                 continue
-            self.process_cr(errors, cr, successes)
+            if patient_unsubscribed:
+                if occurrence_utc < datetime.now(timezone.utc):
```

Comment:
i see elsewhere, we use `datetime.now().astimezone()`
probably the same result, but i'm a fan of consistency.